test/utils.py

```diff
@@ -33,6 +33,12 @@ def assert_tensor_equal(*args, **kwargs):
     torch.testing.assert_close(*args, **kwargs, atol=absolute_tolerance, rtol=0)
 
 
+# Asserts that at most percentage of the elements are different by more than abs_tolerance.
+def assert_tensor_nearly_equal(frame1, frame2, percentage=0.3, abs_tolerance=20):
```

Comment:
Nit regarding the name: we already have `assert_tensor_close`, which semantically convey the same meaning as "nearly equal" to me. So the distinction between these isn't obvious. Maybe `assert_tensor_close_on_at_least(...)`?

Reply:
Agreed. Even better if we can use the same utility function - I suspect that the logic we're doing in this function is quite similar to what `torch.testing.assert_close()` is already doing.

The answer also might be that we just eliminate both `assert_tensor_close()` and `assert_tensor_nearly_equal()`, and just use plain `torch.testing.assert_close()` with scenario-specific tolerances.